### PR TITLE
Improve Supervisor docs and specs

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: lib/elixir/lib/supervisor.ex [hunk 1]
```diff
@@ -529,12 +528,12 @@ defmodule Supervisor do
           | {:max_seconds, pos_integer}
 
   @typedoc "Supported strategies"
-  @type strategy :: :one_for_one | :one_for_all | :rest_for_one
+  @type strategy :: :one_for_one | :one_for_all | :rest_for_one | :simple_one_for_one
```
Comment (eksperimental) [inline lib/elixir/lib/supervisor.ex]: Eventhough it has been deprecated, it should still be reflected in the typespecs

File: lib/elixir/lib/supervisor.ex [hunk 2]
```diff
@@ -529,12 +528,12 @@ defmodule Supervisor do
...
 
   # Note we have inlined all types for readability
   @typedoc "The supervisor specification"
   @type child_spec :: %{
-          required(:id) => term(),
+          required(:id) => module() | term(),
```
Comment (eksperimental) [inline lib/elixir/lib/supervisor.ex]: Even though it's redundant it shows what's the default :id

File: lib/elixir/lib/supervisor.ex (hunk 1 above)
Comment (josevalim) [reply lib/elixir/lib/supervisor.ex]: We have recently agreed in another issue that we won'tkeep in typespecs as we do want the warning there still.

File: lib/elixir/lib/supervisor.ex (hunk 2 above)
Comment (josevalim) [reply lib/elixir/lib/supervisor.ex]: I think `atom()` is better as the `module()` is just a consequence of the atom.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**